src/sentry/types/region.py

```diff
@@ -343,6 +346,21 @@ def find_regions_for_user(user_id: int) -> set[str]:
     return find_regions_for_orgs(org_ids)
 
 
+@control_silo_function
+def find_regions_for_sentry_app(sentry_app: SentryApp) -> set[str]:
+    from sentry.sentry_apps.models.sentry_app_installation import SentryAppInstallation
+
+    if SiloMode.get_current_mode() == SiloMode.MONOLITH:
+        return {settings.SENTRY_MONOLITH_REGION}
+
+    org_ids = set(
+        SentryAppInstallation.objects.filter(sentry_app=sentry_app).values_list(
+            "organization_id", flat=True
+        )
```

Comment:
This can potentially return 1000s of organization IDs. We may want to consider doing the aggregation and org region lookups in a single query if possible, but that means we won't be able to use the `find_regions_for_orgs` helper :/

Reply:
I tried to do this in a single query but failed 🫠  I don't think the Organization model has a foreign key linking to the OrganizationMapping model where the region info is, so I wasn't able to do one long query

Reply:
Organization and OrganizationMapping can't have relations because they are located in different silos. While you can't join sentry_app and organization mapping tables together, you could use a subquery

```python
installation_query = SentryAppInstallation.objects.filter(sentry_app=sentry_app).values_list("organization_id")
regions = OrganizationMapping.objects.filter(organization_id__in=installation_query).distinct("region")
```

This avoids loading the large list of organization_ids into memory and only fetches the region list which will always be  0-2 rows.